src/operations/create_collection.ts

```diff
@@ -96,31 +99,75 @@ export class CreateCollectionOperation extends CommandOperation<Collection> {
     session: ClientSession | undefined,
     callback: Callback<Collection>
   ): void {
-    const db = this.db;
-    const name = this.name;
-    const options = this.options;
+    (async () => {
```

Comment:
Unfortunately we can't use async/await just yet in the driver since we support providing third party promise libs (I know..), I'm sorry I think we have to revert to the callback style. 

@durran am I missing something that makes this API exempt? I looked into FLE and it has its own promise-if-no-callback logic that will use the global promise but I don't think the driver will hit that since it will always pass in a callback to the library

Reply:
@nbbeeken But aren’t third-party promise libs only relevant for the public API? This async function is not externally visible, and even if a client provides a different promise library, they can’t expect that other libraries (or even Node.js itself) don’t use async/await at least *internally*, right?

Reply:
This is a really fair point 🤔  I also thought we could consider that new APIs don't have to abide by the deprecated option. We actually already have a case where snappy 7+ will use native promises, and we callbackify the API.  I'm onboard then. This would be the first time we ship async await in the driver, so lemme get the rest of the team's attention. 2017 here we come :) 

cc: @dariakp @baileympearson